### PR TITLE
TST: add test for getting array from series

### LINKED ISSUE #38543: `np.array([pd.Series({'a':'a'})])` works but `np.array([pd.Series({1:1})])` doesn't
The last line here throws a `KeyError: 0`
```
import numpy as np
import pandas as pd
np.array([pd.Series({0:0})])
np.array([pd.Series({'a':'a'})])
np.array([pd.Series({1:1})])
```
Is that a bug?

### PR DESCRIPTION
**Pull request by horaceklai**
- [x] closes #38543 
- [ x] tests added / passed
- [x ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/arrays/test_array.py [hunk 1]
```diff
@@ -182,6 +182,13 @@ def test_array_copy():
     assert np.shares_memory(a, b._ndarray) is True
 
 
+def test_array_from_series():
```
Comment (jreback) [inline pandas/tests/arrays/test_array.py:185]: can you instead put in pandas/tests/test_constructors.py

and can you assert the results of each of these with `tm.assert_numpy_array_equal`.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/test_array.py [hunk 2]
```diff
@@ -182,6 +182,13 @@ def test_array_copy():
...
+    # GH38543
+    np.array([pd.Series({0: 0})])
```
Comment (mroeschke) [inline pandas/tests/arrays/test_array.py:187]: Could you use `pytest.mark.parameterize` to test each case?

Additionally, these tests should follow the format

```
result = np.array(np.array([pd.Series({0: 0})]))
expected = np.array(...) # another way to construct the result
tm.assert_numpy_array_equal(result, expected)
```

**Comment by mroeschke** [comment]
Mind investigating the failures @horaceklai?

**Comment by horaceklai** [comment]
> Mind investigating the failures @horaceklai?

Do you mean investigating the bug that the issue initially described?

The failure has to do with the bug reoccurring. I thought it was fixed from the comments written before I took this issue, but it is here now.

File: pandas/tests/test_constructors.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import numpy as np
```
Comment (jreback) [inline pandas/tests/test_constructors.py:1]: sorry menat pandas/tests/series/test_constructors.py

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master and ping on green.

**Review by lithomas1 — CHANGES_REQUESTED** [review]
You need to skip this test on old numpy. I believe 1.19 is the first version it works on but haven't checked.

**Comment by horaceklai** [comment]
> You need to skip this test on old numpy. I believe 1.19 is the first version it works on but haven't checked.

Hi Thomas,

I am not sure how to skip certain tests for old numpy. Mind giving me a tip?

Thanks,
Horace

**Comment by lithomas1** [comment]
Have a look at the ``pytest.mark.skipif`` decorator(https://docs.pytest.org/en/latest/reference/reference.html#pytest-mark-skipif-ref). 

You'll also need https://github.com/pandas-dev/pandas/blob/35d52ff2d50ebbf24514407d5986ccf89f56b296/pandas/compat/numpy/__init__.py#L12 from ``pandas.compat.numpy``.
This variable is a boolean that tells you if the numpy version is under 1.19

So you'll have something like 
```
from pandas.compat.numpy import np_version_under1p19
...
...
...
@pytest.mark.skipif(np_version_under1p19, reason="fails on numpy below 1.19")
[your test code]
```

Let me know if you have more questions.

**Comment by horaceklai** [comment]
> Have a look at the `pytest.mark.skipif` decorator(https://docs.pytest.org/en/latest/reference/reference.html#pytest-mark-skipif-ref).
> 
> You'll also need
> 
> https://github.com/pandas-dev/pandas/blob/35d52ff2d50ebbf24514407d5986ccf89f56b296/pandas/compat/numpy/__init__.py#L12
> 
> from `pandas.compat.numpy`.
> This variable is a boolean that tells you if the numpy version is under 1.19
> So you'll have something like
> 
> ```
> from pandas.compat.numpy import np_version_under1p19
> ...
> ...
> ...
> @pytest.mark.skipif(np_version_under1p19, reason="fails on numpy below 1.19")
> [your test code]
> ```
> 
> Let me know if you have more questions.

Thanks! You pretty much did all the work for me and I learned something new!

File: pandas/tests/series/test_constructors.py [hunk 4]
```diff
@@ -1840,3 +1841,17 @@ def test_constructor(rand_series_with_duplicate_datetimeindex):
...
+
+@pytest.mark.parametrize(
+    "input_dict,expected",
+    [
+        ({0: 0}, np.array([[0]], dtype=np.int64)),
+        ({"a": "a"}, np.array([["a"]], dtype=object)),
+        ({1: 1}, np.array([[1]], dtype=np.int64)),
+    ],
+)
+@pytest.mark.skipif(np_version_under1p19, reason="fails on numpy below 1.19")
```
Comment (jreback) [inline pandas/tests/series/test_constructors.py:1854]: can you write a tests that specifically tests < numpy 1.19. e.g. an asserts the error that happens (OT I wonder if this is fixable on the pandas side).

**Comment by horaceklai** [comment]
Yes, good idea.

**Comment by lithomas1** [comment]
@horaceklai Can you fix the test failures?

**Comment by horaceklai** [comment]
> @horaceklai Can you fix the test failures?

I think I fixed it now with the error message not matching, but now I'm having another error that I don't know if it is related. The test is taking too much time (> 1hr)

**Review by lithomas1 — APPROVED** [review]
Lgtm. Thanks for the PR. Don't worry about the timeouts.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @horaceklai

**Pull request merged**